lib/rucio/daemons/reaper/reaper2.py

```diff
@@ -628,7 +639,7 @@ def stop(signum=None, frame=None):
     GRACEFUL_STOP.set()
 
 
-def run(threads=1, chunk_size=100, once=False, greedy=False, rses=None, scheme=None, exclude_rses=None, include_rses=None, vos=None, delay_seconds=0, sleep_time=60, auto_exclude_threshold=100, auto_exclude_timeout=600):
+def run(threads=1, chunk_size=100, once=False, greedy=False, rses=None, scheme=None, exclude_rses=None, include_rses=None, vos=None, delay_seconds=0, sleep_time=60, auto_exclude_threshold=100, auto_exclude_timeout=600, bulk=False):
```

Comment:
A test verifying that the new argument  works as expected would be welcomed. You'll probably need to implement a dummy "bulk_delete()" implementation in  lib/rucio/rse/protocols/mock.py. You could start from one of the simpler tests in lib/rucio/tests/test_reaper2.py

Reply:
To truly mock test this there would need to be some access to a globus service to initiate a transfer then delete.  I will take a look to see how far we could get with this functionally.

Reply:
Truly mocking may be a nice bonus, but is probably complex. However, ensuring that the generic(non-protocol-specific) code inside lib/rucio/daemons/reaper/reaper2.py works as expected should be quite easy to verify.

Reply:
As discussed, would be good to have a mock part here (Not an actual integration test, just mock)